### ISSUE TITLE
-Bug: table odt export ignores relative column width hints 9.4.6 -

### ISSUE THREAD

**Issue by hpgisler** [kjambunathan/org-mode-ox-odt#132]
404

**Comment by QiangF** [comment]
You can download the package manually.
https://github.com/kjambunathan/kjambunathan.github.io/tree/master/elpa

**Comment by hwiorn** [comment]
If you use `straight.el` or `doom-emacs`, you can install `org-mode-ox-odt` using `straight.el` directly. #133

**Comment by kjambunathan** [comment]
@ hpgisler,

I am guessing this bug report is by you https://lists.gnu.org/archive/html/emacs-orgmode/2021-11/msg00220.html.   I see version numbers as 9.3.7.339 there.  So, I believe you got the ELPA stuff working again. 

 I changed the behaviour of how col widths are handled in recent releases.   Column widths now comes from `:col-cookies` or `:widths` line.   Please see these bug reports https://github.com/kjambunathan/org-mode-ox-odt/issues/121, and the related commit https://github.com/kjambunathan/org-mode-ox-odt/commit/cea0effe1e5866ba16bde9bb48b087c2227e29a9#diff-752c3d4750145ca15d5fc71c25f4a8456a6eb7a07eb1cda7458d42bb3c1962d0R5674, specifically the following remarks which addresses the reasoning behind this change 

> Widths may be specified in column cookies.  But those widths aren't what we are looking for here.  Widths specified in column cookies of a table are meant for comfortable viewing/editing of long tables within Emacs by shrinking one or more columns to a manageable width.  These widths are thus functionally different from column widths that one needs in exported documents.  NOTE: The old exporter, specifically the exporter prior to introduction of `:widths' and `:col-cookies' ODT attributes, overloaded widths _within_ the special row of a table to also mean widths of columns in exported document.  This was a convenient hack then.  This hack is no longer needed, and is removed.


@QiangF has indicated that he would prefer the old behaviour to continue.    I am thinking of restoring the old behaviour, or atleast provide an option to revert back to the old behaviour.  Please subscribe to that other bug report to get updates on this front.  Meanwhile, instead of this

```
| <c2> | <l10> |
|------+-------|
|  2   | 10    |
```

you can go with

```
#+ATTR_ODT: :widths "2,10"
| <c>  | <l>   |
|------+-------|
|  2   | 10    |

``` 

or with this 

```
#+ATTR_ODT: :col-cookies "| <c2> | <l10> |"
| <c>  | <l>   |
|------+-------|
|  2   | 10    |

```

I am coping verbatim your report here for some context.



| <c2> | <l10> |
|------+-------|
|  2   | 10    |


#+ATTR_ODT: :widths "2,10"
| <c>  | <l>   |
|------+-------|
|  2   | 10    |


#+ATTR_ODT: :col-cookies "| <c2> | <l10> |"
| <c>  | <l>   |
|------+-------|
|  2   | 10    |


Let the content of a file 'test.org' be the following:

```
  | <c2> | <l10> |
  |------+-------|
  |  2   | 10    |
```


Observed behavior:
test.odt with identical column widths - note, that <c> vs. <l> seems to work

```
 
  |------------------------+-------------------------|
  |            2           | 10                      |

```


Expected behavior:
test.odt width ratio 1/5

```
  |---+-------------------------|
  | 2 | 10                      |
```


Best Reagards
Hanspeter

-----------------------------------------------

Emacs  : GNU Emacs 27.2 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.27, 
cairo version 1.17.4)
 of 2021-03-26
Package: Org mode version 9.4.6 (9.4.6-13-g4be129-elpa @ 
/home/xx/.emacs.d/elpa/org-20210920/)



Output of *Messages*:
--------------------

```
(New file)
Saving file /home/xx/yy/test.org...
Wrote /home/xx/yy/test.org
ODT Zip Dir is /tmp/odt-WAWq3N/
LaTeX to MathML converter not available.
Formatting LaTeX using verbatim
ox-odt: Content template file is 
/home/xx/.emacs.d/elpa/ox-odt-9.3.7.339/etc/styles/OrgOdtContentTemplate.xml
Using schema ~/.emacs.d/elpa/ox-odt-9.3.7.339/etc/schema/od-schema.rnc
Saving file /tmp/odt-WAWq3N/content.xml...
Wrote /tmp/odt-WAWq3N/content.xml
ox-odt: Styles file is 
/home/xx/.emacs.d/elpa/ox-odt-9.3.7.339/etc/styles/OrgOdtStyles.xml
Using schema ~/.emacs.d/elpa/ox-odt-9.3.7.339/etc/schema/od-schema.rnc
Saving file /tmp/odt-WAWq3N/styles.xml...
Wrote /tmp/odt-WAWq3N/styles.xml
Using schema ~/.emacs.d/elpa/ox-odt-9.3.7.339/etc/schema/od-schema.rnc
Saving file /tmp/odt-WAWq3N/meta.xml...
Wrote /tmp/odt-WAWq3N/meta.xml
Wrote /tmp/odt-WAWq3N/mimetype
Using schema ~/.emacs.d/elpa/ox-odt-9.3.7.339/etc/schema/od-manifest-schema.rnc
Saving file /tmp/odt-WAWq3N/META-INF/manifest.xml...
Wrote /tmp/odt-WAWq3N/META-INF/manifest.xml
Create OpenDocument file ‘test.odt’...
Using schema ~/.emacs.d/elpa/ox-odt-9.3.7.339/etc/schema/od-schema.rnc
Running zip -mX0 test.odt mimetype
Running zip -rmTq test.odt .
Created /home/xx/yy/test.odt
Quit
```

**Comment by kjambunathan** [comment]
@hpgisler 

I have responded to your report on `emacs-orgmode` here  https://github.com/kjambunathan/org-mode-ox-odt/issues/132#issuecomment-966826127.

If you are using my version of the exporter, the right forum to report the issues will be on this github repo.    As the [README](https://github.com/kjambunathan/org-mode-ox-odt#readme) says, my version of `ox-odt` is a FORK of the original exporter you get from installing Emacs, and the features (and bugs) I introduce will NEVER be part of upstream Emacs.

**Comment by kjambunathan** [comment]
> https://lists.gnu.org/archive/html/emacs-orgmode/2021-11/msg00220.html

@hpgisler, the fix you are looking for is included in `elpa/ox-odt-9.3.7.361.tar `.  Btw, please report any issues with this repository, right in this repository (and not in `emacs-orgmode` or any other `emacs` official lists).   The ODT exporter is 11 years old, and for all practical purposes I have been the sole contributor.  And ... the ODT exporter in vanilla Emacs is in the same state as I had left it very many years ago.  If you care about fixes and continued support you should be raising issues here and not in any of the official Emacs list.

**Comment by hpgisler** [comment]
@kjambunathan Thank you so very much for your help and effort. 
It now works again great - and sorry for my ignorance.
